data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java

```diff
@@ -19,10 +19,22 @@
 
     private AadAuthenticationHelper aadAuthenticationHelper;
     private String clusterUrl;
+    private String clientVersionForTracing;
+    private String applicationNameForTracing;
 
     public ClientImpl(ConnectionStringBuilder csb) throws URISyntaxException {
         clusterUrl = csb.getClusterUrl();
         aadAuthenticationHelper = new AadAuthenticationHelper(csb);
+        clientVersionForTracing = "Kusto.Java.Client";
+        String version = Utils.GetPackageVersion();
+        if (StringUtils.isNotBlank(version)) {
+            clientVersionForTracing += ":" + version;
+        }
+
+        if (StringUtils.isNotBlank(csb.getClientVersionForTracing())) {
```

Comment:
Do we still need that if the user is going to use the applicationNameForTracing ?

Reply:
Yes. We need to modify it from the ingest client.